sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -165,6 +167,33 @@ class Analyzer(
     }
   }
 
+  object ValidateGenerator extends Rule[LogicalPlan] {
+    private def hasGenerator(expr: Expression): Boolean = {
+      expr.find(_.isInstanceOf[Generator]).isDefined
+    }
+
+    private def hasNestedGenerator(expr: NamedExpression): Boolean = expr match {
+      case UnresolvedAlias(_: Generator, _) => false
+      case Alias(_: Generator, _) => false
+      case MultiAlias(_: Generator, _) => false
+      case other if hasGenerator(other) => true
+      case _ => false
+    }
+
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case Project(projectList, _) if projectList.exists(hasNestedGenerator) =>
+        throw new AnalysisException("Generators are not supported when it's nested in expressions.")
```

Comment:
i think we need to show the generator itself, as "generator" is a very spark sql specific term. If users just put an explode, he/she wont' know it's a generator.